### PR TITLE
proxmox-image: add additionalDiskSpace parameter as input to make-disk-image.nix

### PR DESCRIPTION
**Pull request by jakeschurch**
## Things done

Add additionalDiskSpace option for proxmox images to be passed to make-disk-image.nix.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/proxmox-image.nix [hunk 1]
```diff
@@ -94,6 +94,15 @@ with lib;
...
       };
+
+      additionalDiskSize = mkOption {
+        type = types.str;
+        default = "512M";
+        description = lib.mdDoc ''
+          Additional disk space to be added to the image.
+          Defaults to 512M (Megabytes), Suffix can also be specified with `G` (gigabyte) or `K` (kilobyte).
+        '';
+      };
```
Comment (drupol) [inline nixos/modules/virtualisation/proxmox-image.nix:105]:
```suggestion
      additionalDiskSize = lib.mkOption {
        type = lib.types.str;
        default = "512M";
        description = lib.mdDoc ''
          Additional disk space to be added to the image.
          Defaults to 512M (Megabytes), Suffix can also be specified with `G` (gigabyte) or `K` (kilobyte).
        '';
      };
```

File: nixos/modules/virtualisation/proxmox-image.nix [hunk 2]
```diff
@@ -117,6 +126,14 @@ with lib;
...
       example = "hybrid";
     };
+    additionalDiskSpace = mkOption {
+      type = types.str;
+      default = "512M";
+      description = lib.mdDoc ''
+        Additional disk space to be added to the image.
+        Defaults to 512M (Megabytes), Suffix can also be specified with `G` (gigabyte) or `K` (kilobyte).
+        '';
+    };
```
Comment (drupol) [inline nixos/modules/virtualisation/proxmox-image.nix:136]:
```suggestion
    additionalDiskSpace = lib.mkOption {
      type = lib.types.str;
      default = "512M";
      description = lib.mdDoc ''
        Additional disk space to be added to the image.
        Defaults to 512M (Megabytes), Suffix can also be specified with `G` (gigabyte) or `K` (kilobyte).
        '';
    };
```

File: nixos/modules/virtualisation/proxmox-image.nix [hunk 3]
```diff
@@ -168,6 +185,7 @@ with lib;
     system.build.VMA = import ../../lib/make-disk-image.nix {
       name = "proxmox-${cfg.filenameSuffix}";
       inherit partitionTableType;
+      additionalSpace = config.proxmox.additionalDiskSpace;
```
Comment (drupol) [inline nixos/modules/virtualisation/proxmox-image.nix:188]: How about naming the configuration option `additionalSpace` and then do `inherit partitionTableType additionalSpace;` ?

**Review by drupol — COMMENTED** [review]
Hi ! 

Thanks for your first contribution to NixOS.

I've left a few comments, let me know if it's ok for you to process them.

Thanks!

**Review by drupol — APPROVED** [review]

**Comment by jakeschurch** [comment]
thanks @drupol!

**Pull request merged**